pkg/controllers/management/kontainerdrivermetadata/metadata_handler.go

```diff
@@ -98,27 +103,25 @@ func (m *MetadataController) sync(key string, test *v3.Setting) (runtime.Object,
 	return test, nil
 }
 
-func (m *MetadataController) initTicker(ctx context.Context) {
-	url, interval, err := getSettingValues()
-	if err != nil {
-		panic(fmt.Errorf("driverMetadata error getting interval in int %s %v", settings.RkeMetadataRefreshIntervalMins.Get(), err))
-	}
+func (m *MetadataController) initTicker(ctx context.Context, interval time.Duration) {
 	cctx, cancel := context.WithCancel(ctx)
-
-	tickerData = &TickerData{cancelFunc: cancel, url: url, interval: interval}
-
+	tickerData = &TickerData{cancelFunc: cancel, interval: interval}
 	go m.startTicker(cctx, tickerData, true)
 }
 
 func (m *MetadataController) startTicker(ctx context.Context, tickerData *TickerData, init bool) {
-	checkInterval, url := tickerData.interval, tickerData.url
+	checkInterval := tickerData.interval
 	for range ticker.Context(ctx, checkInterval) {
 		logrus.Infof("driverMetadata: checking rke-metadata-url every %v", checkInterval)
-		m.refresh(url, false)
+		url, _, err := GetSettingValues()
+		if err != nil {
+			logrus.Errorf("driverMetadata: error getting settings %v", err)
+		}
+		m.Refresh(url, false)
 	}
 }
 
-func (m *MetadataController) refresh(url string, init bool) {
+func (m *MetadataController) Refresh(url string, init bool) {
```

Comment:
I think we should add an error as a return value. And then let the caller decide on what to do with it.